clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6118,6 +6120,10 @@ void Sema::warnOnReservedIdentifier(const NamedDecl *D) {
     Diag(D->getLocation(), diag::warn_reserved_extern_symbol)
         << D << static_cast<int>(Status);
   }
+  // Diagnose use of C++ keywords in C as being incompatible with C++.
+  if (const IdentifierInfo *II = D->getIdentifier();
```

Comment:
Is there a reason to warn on this here rather than in the same place we warn on "future keywords"? (When the preprocessor produces the final identifier token.) I would worry that this check would miss some cases. (Parameter names appear to need special handling, and they're probably not going to remain the only case even if they currently are.)

Reply:
I was thinking this was similar to reserved identifiers, but I think your suggestion is cleaner. I'll work on an update for that